pygmsh/built_in/geometry.py

```diff
@@ -492,6 +513,16 @@ def add_rectangle(
             make_surface=make_surface
             )
 
+    class Polygon(object):
+        def __init__(self, line_loop, surface, lcar):
+            if hasattr(surface, 'id'):
+                self.id = surface.id
+            self.line_loop = line_loop
+            self.num_edges = len(line_loop.lines)
```

Comment:
`num_edges` is redundant.

Reply:
this is required to make a `Polygon` work with `extrude()`, as `SurfaceBase` does not have a real `LineLoop` inside and we have to resort to the `num_edges` attribute over there.

Reply:
I will remove also the `id`. I figured I can pass `poly.surface` directly.